tests/test_flow/test_reload.py

```diff
@@ -0,0 +1,257 @@
+import pytest
+
+import sys
+from textwrap import dedent
+
+from bionic.exception import CodeVersioningError
+
+
+# Pytest rewrites modules and caches the compiled pyc file. The cache can
+# interfere with the tests below. Hence, we disable the cache here.
+@pytest.fixture(autouse=True)
+def dont_write_bytecode():
+    dont_write_bytecode = sys.dont_write_bytecode
+    yield
+    sys.dont_write_bytecode = dont_write_bytecode
+
+
+@pytest.fixture
+def tmp_module_path(tmp_path):
+    tmp_dir_name = str(tmp_path)
+    sys.path.insert(0, tmp_dir_name)
+
+    sys.dont_write_bytecode = True
+
+    yield tmp_path
+
+    # Ensure that the temporary module is not loaded by a future test.
+    sys.path.remove(tmp_dir_name)
+
+
+@pytest.fixture
+def write_flow_file(tmp_module_path):
```

Comment:
From the names `write_flow_file` and `write_flow_module`, I'd guess that they do related things, but one deals with *files* and the other deals with *modules*. However, the actual difference is that one deals with *arbitrary* modules and the other deals with *flow* modules. Maybe a better name for this function would be `write_module`?

Reply:
Renamed module writing function to `write_module` and flow writing function to `write_flow_module`.